test/e2e/preflight_permission_validation_escalation_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func TestPreflightPermissionValidationEscalation(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	testName := "preflight-permission-validation-escalation"
+
+	// Create the minimal resources necessary for testing kapp
+	base := `
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: __test-name__
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: scoped-sa
+  namespace: __ns__
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: scoped-sa
+  namespace: __ns__
+  annotations:
+    kubernetes.io/service-account.name: scoped-sa
+type: kubernetes.io/service-account-token
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: __test-name__
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: ["*"]
+- apiGroups: [""]
+  resources: ["namespaces"]
+  verbs: ["list"]
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "list"]
+- apiGroups: ["rbac.authorization.k8s.io"]
+  resources: ["roles", "rolebindings", "clusterroles", "clusterrolebindings"]
+  verbs: ["get", "list", "create", "update", "delete", "escalate", "bind"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: __test-name__
+subjects:
+- kind: ServiceAccount
+  name: scoped-sa
+  namespace: __ns__
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: __test-name__
+`
+
+	base = strings.ReplaceAll(base, "__test-name__", testName)
+	base = strings.ReplaceAll(base, "__ns__", env.Namespace)
+	baseName := "preflight-permission-validation-base-app"
+	appName := "preflight-permission-validation-app"
+	scopedContext := "scoped-context"
+	scopedUser := "scoped-user"
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", baseName})
+		kapp.Run([]string{"delete", "-a", appName})
+		RemoveClusterResource(t, "ns", testName, "", kubectl)
+	}
+	cleanUp()
+	defer cleanUp()
+
+	kapp.RunWithOpts([]string{"deploy", "-a", baseName, "-f", "-"}, RunOpts{StdinReader: strings.NewReader(base)})
+	cleanUpContext := ScopedContext(t, kubectl, testName, scopedContext, scopedUser)
+	defer cleanUpContext()
+
+	// Verify that permissions are missing for applying a (Cluster)Role
```

Comment:
Could you update these comments. They are saying "missing" for all the tests cases.

Reply:
Should we also add a failing test case for escalation?

Reply:
In the latest push I:
- Removed the comments from all the tests as they aren't really necessary and I had originally used them as a check list for what I needed to implement
- Added an e2e test for failing privilege escalation